src/poisson.f90

```diff
@@ -298,89 +298,123 @@ subroutine poisson_000(rhs)
-                             tmp2 * bx(i) - tmp1 * ax(i), kind=mytype)
-             if (i > (nx/2+1)) cw1(i,j,k) = -cw1(i,j,k)
-
-             ! Solve Poisson
-             tmp1 = real(kxyz(i,j,k), kind=mytype)
-             tmp2 = aimag(kxyz(i,j,k))
-             ! CANNOT DO A DIVISION BY ZERO
-             if ((tmp1 < epsilon).or.(tmp2 < epsilon)) then
-                cw1(i,j,k) = zero
-             else
-                cw1(i,j,k) = cmplx(real(cw1(i,j,k),kind=mytype) / (-tmp1), &
-                                  aimag(cw1(i,j,k)) / (-tmp2), kind=mytype)
-             end if
-
-             !Print result in spectal space after Poisson
-             !     if (abs(out(i,j,k)) > 1.0e-4) then
-             !        write(*,*) 'AFTER',i,j,k,out(i,j,k),xyzk
-             !     end if
-
-             ! post-processing backward
-
-             ! POST PROCESSING IN Z
-             tmp1 = real(cw1(i,j,k), kind=mytype)
-             tmp2 = aimag(cw1(i,j,k))
-             cw1(i,j,k) = cmplx(tmp1 * bz(k) - tmp2 * az(k), &
-                            -tmp2 * bz(k) - tmp1 * az(k), kind=mytype)
-
-             ! POST PROCESSING IN Y
-             tmp1 = real(cw1(i,j,k), kind=mytype)
-             tmp2 = aimag(cw1(i,j,k))
-             cw1(i,j,k) = cmplx(tmp1 * by(j) + tmp2 * ay(j), &
-                             tmp2 * by(j) - tmp1 * ay(j), kind=mytype)
-             if (j > (ny/2 + 1)) cw1(i,j,k) = -cw1(i,j,k)
-
-             ! POST PROCESSING IN X
-             tmp1 = real(cw1(i,j,k), kind=mytype)
-             tmp2 = aimag(cw1(i,j,k))
-             cw1(i,j,k) = cmplx(tmp1 * bx(i) + tmp2 * ax(i), &
-                            -tmp2 * bx(i) + tmp1 * ax(i), kind=mytype)
-             if (i > (nx/2+1)) cw1(i,j,k) = -cw1(i,j,k)
-       ! post-processing in spectral space
-
+      ! POST PROCESSING IN Z
+      cw1_tmp = cw1(i,j,k)
+      tmp1 = real(cw1_tmp, kind=mytype)/ntot
+      tmp2 = aimag(cw1_tmp)/ntot
+      cw1_tmp = cmplx(tmp1 * bz(k) + tmp2 * az(k), &
+                      tmp2 * bz(k) - tmp1 * az(k), kind=mytype)
+      
+      ! POST PROCESSING IN Y
+      tmp1 = real(cw1_tmp, kind=mytype)
+      tmp2 = aimag(cw1_tmp)
+      cw1_tmp = cmplx(tmp1 * by(j) + tmp2 * ay(j), &
+                      tmp2 * by(j) - tmp1 * ay(j), kind=mytype)
+      if (j > (ny/2+1)) cw1_tmp = -cw1_tmp
+      
+      ! POST PROCESSING IN X
+      tmp1 = real(cw1_tmp, kind=mytype)
+      tmp2 = aimag(cw1_tmp)
+      cw1_tmp = cmplx(tmp1 * bx(i) + tmp2 * ax(i), &
+                      tmp2 * bx(i) - tmp1 * ax(i), kind=mytype)
+      if (i > (nx/2+1)) cw1_tmp = -cw1_tmp
+      
+      ! Solve Poisson
+      tmp1 = real(kxyz(i,j,k), kind=mytype)
+      tmp2 = aimag(kxyz(i,j,k))
+      ! CANNOT DO A DIVISION BY ZERO
+      if ((tmp1 < epsilon).or.(tmp2 < epsilon)) then
+         cw1_tmp = zero
+      else
+         cw1_tmp = cmplx(real(cw1_tmp,kind=mytype) / (-tmp1), &
+                           aimag(cw1_tmp) / (-tmp2), kind=mytype)
```

Comment:
The formatting seems exotic, have you applied fprettify ?

Reply:
Not yet. I wanted to merge and after apply fpreffify